src/EventStore.Core/TransactionLog/FileNamingStrategy/VersionedPatternFileNamingStrategy.cs

```diff
@@ -44,6 +45,73 @@ public class VersionedPatternFileNamingStrategy : IFileNamingStrategy {
 			return versions;
 		}
 
+		private int GetFileNumberFor(string path) {
+			var fileName = Path.GetFileName(path);
+			if (!_chunkNamePattern.IsMatch(fileName))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			var start = _prefix.Length;
+			var end = fileName.IndexOf('.', _prefix.Length);
+			Debug.Assert(end != -1);
+
+			if (!int.TryParse(fileName[start..end], out var chunkNumber))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			return chunkNumber;
+		}
+
+		private int GetFileVersionFor(string path) {
+			var fileName = Path.GetFileName(path);
+			if (!_chunkNamePattern.IsMatch(fileName))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			var dot = fileName.IndexOf('.', _prefix.Length);
+			Debug.Assert(dot != -1);
+
+			if (!int.TryParse(fileName[(dot+1)..], out var chunkVersion))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			return chunkVersion;
```

Comment:
This is subtly different from before, which is probably ok. The regex allows any character in the version part of the file name and I am not sure I can think of a good reason for it.

Reply:
Would it be a good time to update the regex to accept only digits? Since this part of the code makes the assumption that it should be a number.

Reply:
Resolved in 2784d510d333a70dc0996184701cc08fa238bca9